src/Symfony/Component/Mailer/Bridge/Mailgun/Http/Api/MailgunTransport.php

```diff
@@ -58,6 +58,10 @@ protected function doSendEmail(Email $email, SmtpEnvelope $envelope): void
         ]);
 
         if (200 !== $response->getStatusCode()) {
+            if (isset($response->getHeaders()['content-type'][0]) && !preg_match('/\bjson\b/i', $response->getHeaders()['content-type'][0])) {
+                throw new TransportException(sprintf('Unable to send an email (code %s).', $response->getStatusCode()));
+            }
+
             $error = $response->toArray(false);
```

Comment:
I would instead wrap this is a try/catch and rethrow in the catch.
But when does this happen? This should be investigated on the original issue first to me.